middleware/getQuery.js

```diff
@@ -1,5 +1,8 @@
 exports.getSearchObject = (reqBody) => {
-  const isEmail = reqBody.email.match(/^\w+@[a-zA-Z_]+?\.[a-zA-Z]{2,3}$/);
+  const isEmail = reqBody.email.match(
+    // eslint-disable-next-line no-useless-escape
+    /^\w+[\w-\.]*\@\w+((-\w+)|(\w*))\.[a-z]{2,3}$/
```

Comment:
Could you please change this regex to this one
> /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/

As the one you added here does not validate for all test cases

This what your regex produces
![image](https://user-images.githubusercontent.com/61190974/146183144-b0aae206-576c-4f20-ba19-8e75f2dafec4.png)


and this is what it should produce
![image](https://user-images.githubusercontent.com/61190974/146182998-a5a77bf6-a8fd-475f-a785-6e5322131677.png)

Reply:
done but remove the joi.email because user could enter phone number or email  
![image](https://user-images.githubusercontent.com/61456270/147828493-41a961af-1c1e-4466-b1ab-cbe0cfbbb223.png)